src/main/java/edu/cornell/kfs/sys/service/impl/DocumentMaintenanceServiceImpl.java

```diff
@@ -1,45 +1,100 @@
 package edu.cornell.kfs.sys.service.impl;
 
+import edu.cornell.kfs.sys.CUKFSConstants;
+import edu.cornell.kfs.sys.dataaccess.ActionItemNoteDetailDto;
 import edu.cornell.kfs.sys.dataaccess.DocumentMaintenanceDao;
 import edu.cornell.kfs.sys.service.DocumentMaintenanceService;
+
+import org.apache.commons.collections.CollectionUtils;
+import org.apache.commons.lang3.StringUtils;
 import org.apache.logging.log4j.LogManager;
 import org.apache.logging.log4j.Logger;
-import org.kuali.rice.core.api.config.CoreConfigHelper;
-import org.kuali.rice.kew.api.KewApiServiceLocator;
+import org.kuali.kfs.krad.util.ObjectUtils;
+import org.kuali.kfs.sys.KFSPropertyConstants;
+import org.kuali.kfs.sys.context.SpringContext;
+import org.kuali.rice.core.api.criteria.PredicateFactory;
+import org.kuali.rice.core.api.criteria.QueryByCriteria;
+import org.kuali.rice.kew.actionitem.ActionItem;
+import org.kuali.rice.kew.actionitem.ActionItemExtension;
 import org.kuali.rice.kew.api.document.DocumentRefreshQueue;
+import org.kuali.rice.krad.service.KRADServiceLocator;
 import org.springframework.transaction.annotation.Transactional;
 
+import java.util.ArrayList;
 import java.util.Collection;
-import java.util.Iterator;
+import java.util.List;
 
 public class DocumentMaintenanceServiceImpl implements DocumentMaintenanceService {
     private static final Logger LOG = LogManager.getLogger(DocumentMaintenanceServiceImpl.class);
-
-    public static final int WAIT_TIME_NO_WAIT = 0;
-
     private DocumentMaintenanceDao documentMaintenanceDao;
 
     @Transactional
     public boolean requeueDocuments() {
         Collection<String> docIds = documentMaintenanceDao.getDocumentRequeueValues();
         LOG.info("requeueDocuments: Total number of documents flagged for requeuing: " + docIds.size());
 
+        List<ActionItemNoteDetailDto> noteDetails = documentMaintenanceDao.getActionNotesToBeRequeued();
+        LOG.info("requeueDocuments: Total number of action note details: " + noteDetails.size());
+        
         for (String docId : docIds) {
             LOG.info("requeueDocuments: Requesting requeue for document: " + docId);
-            DocumentRefreshQueue documentRequeuer = KewApiServiceLocator.getDocumentRequeuerService(CoreConfigHelper.getApplicationId(), docId, WAIT_TIME_NO_WAIT);
+            DocumentRefreshQueue documentRequeuer = (DocumentRefreshQueue) SpringContext.getService(CUKFSConstants.DOCUMENT_REFRESH_QUEUE_SERVICE_SPRING_CONTEXT_NAME);
             documentRequeuer.refreshDocument(docId);
+            
+            List<ActionItemNoteDetailDto> noteDetailsForDocument = findNoteDetailsForDocument(noteDetails, docId);
+            for (ActionItemNoteDetailDto detailDto : noteDetailsForDocument) {
+                List<ActionItem> actionItems = findActionItem(detailDto);
+                if (CollectionUtils.isNotEmpty(actionItems)) {
+                    ActionItem item = actionItems.get(0);

```

Comment:
Users can potentially have multiple action items on the same document; not sure how likely that would happen in this requeue case. You may want to force the code to work with the user's highest-priority action item for the document, to ensure that the note doesn't get added to the wrong item. Such prioritization can be handled by Rice's ActionListPriorityComparator class, and by Cynergy's ActionListBuilder.ActionListPriorityAndIdComparator class. (The latter might be preferable because of our Cynergy action list customizations.)

Alternately, if this requeue is only happening in a very limited case where a user would not have multiple action items (not even a "READ NOTES" FYI in addition to APPROVE), then such prioritization-grabbing might not be needed.

Reply:
Currently we are only requeueing fiscal officer action items.  I haven't seen multiple action items for a single person/document in my local testing,  Let me see if I can the priority processing you mentioned